Classes/Domain/Model/Category.php

```diff
@@ -44,15 +44,15 @@ public function setDescription($description): void
         $this->description = $description;
     }
 
-    public function getParent(): Category
+    public function getParent(): ?Category
     {
         if ($this->parent instanceof LazyLoadingProxy) {
             $this->parent->_loadRealInstance();
         }
         return $this->parent;
     }
 
-    public function setParent(Category $parent): void
+    public function setParent(?Category $parent = null): void
```

Comment:
I would remove the default.

```suggestion
    public function setParent(?Category $parent): void
```